### PR TITLE
(feat): CSV download component

### PR DESCRIPTION
**Pull request by kieckhafer**
A simple component to download data in a CSV format.

Link to docs PR: https://github.com/reactioncommerce/reaction-docs/pull/427

### CODE REVIEW

File: imports/plugins/core/ui/client/components/csvDownload/csvDownload.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import PropTypes from "prop-types";
+import { Components, registerComponent } from "@reactioncommerce/reaction-components";
+import { CSVLink } from "react-csv";
+
+/**
+ * @name CSVDownload propTypes
+ * @summary React component for downloading CSV file of provided data, a wrapper around `react-csv`
+ * @module CSVDownload
+ * @extends Component
+ * @file CSVDownalod React Component for downloading data in CSV Format
```
Comment (machikoyasuda) [inline imports/plugins/core/ui/client/components/csvDownload/csvDownload.js]: `CSVDownload`

**Comment by pmn4** [comment]
lgtm

**Comment by zenweasel** [comment]
@kieckhafer Just noticed that the CSV library you added has a reported vulnerability (same shell.js vulnerability that Jeremy eliminated in the CLI)

**Comment by spencern** [comment]
This is a great feature, but I'd like to get the vulnerability resolved before merging it into a release. Holding off for now.

**Comment by zenweasel** [comment]
@jshimko and I looked into this vulnerability over the weekend and it's that the `documentation` package relies on shelljs to perform just two functions that can be replaced with native Node code. I think Jeremy was going to try and put a PR in to the original project, but fixing this would depending on both `documentation` and `react-csv` accepting upstream changes and updating.

**Comment by kieckhafer** [comment]
I'd rate this as a "nice-to-have", nothing in the app currently uses this component, so if there are unsolved security issues that we can't do anything about, putting this on hold wouldn't break anything.

**Comment by spencern** [comment]
I think this is ready to go whenever the downstream dependencies get their security issues resolved, but as it's more of a "nice-to-have" I'm going to icebox it for now.

**Comment by zenweasel** [comment]
We do have a WIP Export Order CSV plugin that this would probably in which this would probably be helpful. https://github.com/reactioncommerce/reaction-export-orders-csv

**Comment by spencern** [comment]
@kieckhafer sometime this week let's look at this and see if the deps have been updated yet

**Comment by aaronjudd** [comment]
Let's close this, suggest that it should move to `reaction-component-library`?  Re-open if anyone disagrees, but as this is not actually implemented and is more of a lib item....

**Pull request closed without merging**